core/query/sql.ml

```diff
@@ -18,8 +18,11 @@ type query =
       upd_where: base option
     }
   | Delete    of { del_table: string; del_where: base option }
-  | With      of Var.var * query * Var.var * query
-and select_clause = (base * string) list * (string * Var.var) list * base * base list
+  | With      of string * query * query
```

Comment:
I'm not super happy with this either, since (at least in principle) names of tables introduced by WITH are bound and should alpha-vary.  However, everywhere else where we work with table names (e.g. `TableRef`, `Insert`, `Delete`, `Update` they are raw strings, and it isn't sensible to turn all of these into variables (since in the IR, variables are just numbers).  For the variables in WITH and their occurrences, we currently turn these into unique strings when constructing the WITH expression in `query.ml`.

For the moment I'll introduce a type alias for table names so that it's clear that even though they are currently defined as strings, we might want to do something else in the future such as allow either a string constant (i.e. name of a base table or view defined in the DB) or a variable (name of a table introduced by With).